[HDInsight] Add three new features GetCapabilites, AutoScale, GetBillingSpec

aim-for-better opened the pull request:
The BillingSpec feature is related to Spec PR  [[HDInsight] Add ListBillingSpecs API to HDInsight's stable and preview version](https://github.com/Azure/azure-rest-api-specs/pull/6360)

The Capabilities feature is related to Spec PR[ [HDInsight] Add Capabilities to Stable API version](https://github.com/Azure/azure-rest-api-specs/pull/6665)

The autoscale feature is related to Spec PR [HDI add autoscale feature support ](https://github.com/Azure/azure-rest-api-specs/pull/6158)

1. Add new feature GetCapabilites, AutoScale, GetBillingSpec to HDInsight SDK
2. Add Capability Test, AutoScale Test
3. Update Version from 5.0.0 to 5.1.0
4. Update release notes

Commit:
Changes:
1.Add new feature GetCapabilites, AutoScale, GetBillingSpec to HDInsight SDK
2.Add Capability Test, AutoScaleTest
3.Update Version from 5.0.0 to 5.1.0
4.Update release notes
5. regenerate SDK use generate.ps1 and add eng\mgmt\mgmtmetadata\hdinsight_resource-manager.txt
   and SdkInfo_HDInsightManagementClinet.cs

isra-fel reviewed (changes requested):
Hi @aim-for-better , could you please
1. Add a link to the spec PR in the discription,
1. and follow the [instrcutions here](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec) to regenerate SDK. (remember to git add `eng\mgmt\mgmtmetadata\<rpName>.txt`)

aim-for-better commented:
> Hi @aim-for-better , could you please
> 
> 1. Add a link to the spec PR in the discription,
> 2. and follow the [instrcutions here](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec) to regenerate SDK. (remember to git add `eng\mgmt\mgmtmetadata\<rpName>.txt`)

Hi @isra-fel,
Thanks for your comments. I have added some spec PRs , regenerated the SDK and added some files to the repo.

isra-fel reviewed (approved):
It looks good to me.
